src/pages/[platform]/build-a-backend/auth/connect-your-frontend/switching-authentication-flows/index.mdx

```diff
@@ -105,6 +113,99 @@ The flow is initiated by calling `signIn` with `AuthSignInOptions` configured wi
+
+### Migrate users with Amazon Cognito
+
+Amazon Cognito provides a trigger to migrate users from your existing user directory seamlessly into Cognito. You achieve this by configuring your User Pool's "Migration" trigger which invokes a Lambda function whenever a user that does not already exist in the user pool authenticates, or resets their password.
+
+In short, the Lambda function will validate the user credentials against your existing user directory and return a response object containing the user attributes and status on success. An error message will be returned if an error occurs. Visit [Amazon Cognito user pools import guide](https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pools-import-using-lambda.html) for migration flow and more detailed instruction, and [Amazon Cognito Lambda trigger guide](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-migrate-user.html#cognito-user-pools-lambda-trigger-syntax-user-migration) on how to set up lambda to handle request and response objects.
+
+## `CUSTOM_WITH_SRP` & `CUSTOM_WITHOUT_SRP` flows
+
+Amazon Cognito user pools supports customizing the authentication flow to enable custom challenge types, 
+in addition to a password in order to verify the identity of users. These challenge types may include CAPTCHAs
+or dynamic challenge questions. The `CUSTOM_WITH_SRP` flow requires a password when calling `signIn`. Both of 
+these flows map to the `CUSTOM_AUTH` flow in Cognito.
+
+To define your challenges for custom authentication flow, you need to implement three Lambda triggers for Amazon Cognito.
+
+<Callout>
+
+For more information about working with Lambda Triggers for custom authentication challenges, please visit [Amazon Cognito Developer Documentation](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-challenge.html).
+
+</Callout>
+
+### Custom authentication flow
+
+To initiate a custom authentication flow in your app, call `signIn` without a password. A custom challenge needs to be answered using the `confirmSignIn` API:
+
+```ts title="amplify/auth/handler.ts"
+import { signIn, confirmSignIn } from 'aws-amplify/auth';
+
+export async function handleSignIn(username: string) {
+  const challengeResponse = 'the answer for the challenge';
+  try {
+    const { nextStep } = await signIn({
+      username,
+      options: {
+        authFlowType: 'CUSTOM_WITHOUT_SRP',
+      },
+    });
+    if (nextStep.signInStep === 'CONFIRM_SIGN_IN_WITH_CUSTOM_CHALLENGE') {
+      try {
+        // to send the answer of the custom challenge
+        const output = await confirmSignIn({ challengeResponse });
+        console.log(output);
+      } catch (err) {
+        console.log(err);
+      }
+    }
+  } catch (err) {
+    console.log(err);
+  }
+}
+
+```
```

Comment:
````suggestion
```ts title="src/main.ts"
import { signIn, confirmSignIn } from 'aws-amplify/auth';

const challengeResponse = 'the answer for the challenge';

const { nextStep } = await signIn({
  username,
  options: {
    authFlowType: 'CUSTOM_WITHOUT_SRP',
  },
});

if (nextStep.signInStep === 'CONFIRM_SIGN_IN_WITH_CUSTOM_CHALLENGE') {
  // to send the answer of the custom challenge
  await confirmSignIn({ challengeResponse });
}
```
````

filename for this snippet will be some client-side file, also reduced the snippet to showcase the API

Reply:
Got it, thank you for correcting it 👍